engine/runtime/src/main/java/org/enso/interpreter/caches/ModuleCache.java

```diff
@@ -8,66 +8,74 @@
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
+import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Optional;
-import java.util.UUID;
 import java.util.logging.Level;
 import org.apache.commons.lang3.StringUtils;
 import org.enso.compiler.core.ir.Module;
-import org.enso.compiler.core.ir.ProcessingPass;
 import org.enso.interpreter.runtime.EnsoContext;
 import org.enso.interpreter.runtime.builtin.Builtins;
 import org.enso.persist.Persistance;
 import org.enso.polyglot.CompilationStage;
 
-public final class ModuleCache extends Cache<ModuleCache.CachedModule, ModuleCache.Metadata> {
-
+public final class ModuleCache
+    implements Cache.Spi<ModuleCache.CachedModule, ModuleCache.Metadata> {
```

Comment:
`Cache` is not `final` class. All the cache implementations now implement the `Spi` interface and create instance of `Cache` with its `create` method.